data/json/mapgen/dairy_farm.json

```diff
@@ -0,0 +1,251 @@
+[
+  {
+    "type":"monstergroup",
+    "name" : "GROUP_COWS",
+    "is_safe" : true,
+    "default" : "mon_null",
+    "monsters":[{"monster" : "mon_cow", "cost_multiplier" : 10, "freq": 100, "ends" : 168 },
+    {"monster" : "mon_cow", "freq": 50, "cost_multiplier" : 5, "starts" :  336  }
+    ]
+  },{
+    "method": "json",
+    "object": {
+    "place_monsters": [
+        { "monster": "GROUP_COWS", "x": [ 1, 10 ], "y": [ 1, 9 ], "density": 0.5 },
+        { "monster": "GROUP_COWS", "x": [ 1, 12 ], "y": [ 1, 12 ], "density": 0.5 },
+        { "monster": "GROUP_COWS", "x": [ 1, 11 ], "y": [ 1, 10 ], "density": 0.5 },
+        { "monster": "GROUP_COWS", "x": [ 1, 9 ], "y": [ 1, 11 ], "density": 0.5 },
+        { "monster": "GROUP_COWS", "x": [ 1, 15 ], "y": [ 1, 14 ], "density": 0.5 }
+
+      ],
+        "rows": [
+            "........................",
+            ".$$$$$$$$$$$$$$$$$$$$$$$",
+            ".$......................",
+            ".$......................",
+            ".$......................",
+            ".$......................",
+            ".$......................",
+            ".$......................",
+            ".$......................",
+            ".$......................",
+            ".$......................",
+            ".$......................",
+            ".$......................",
+            ".$......................",
+            ".$......................",
+            ".$......................",
+            ".$......................",
+            ".$......................",
+            ".$......................",
+            ".$......................",
+            ".$......................",
+            ".$......................",
+            ".$......................",
+            ".$......................"
+        ],
+        "terrain": {
+            "$": "t_chainfence_v",
+            ".": ["t_grass", "t_grass", "t_dirt", "t_dirt", "t_dirt"]
+        }
+    },
+    "om_terrain": [ "dairy_farm_NW" ],
+    "type": "mapgen",
+    "weight": 250
+},
+  {
+    "method": "json",
+    "object": {
+        "furniture": {
+            "$": "f_null",
+            ".": "f_null"
+        },
+        "place_monsters": [
+        { "monster": "GROUP_COWS", "x": [ 1, 10 ], "y": [ 1, 9 ], "density": 0.5 },
+        { "monster": "GROUP_COWS", "x": [ 1, 12 ], "y": [ 1, 12 ], "density": 0.5 },
+        { "monster": "GROUP_COWS", "x": [ 1, 11 ], "y": [ 1, 10 ], "density": 0.5 },
+        { "monster": "GROUP_COWS", "x": [ 1, 9 ], "y": [ 1, 11 ], "density": 0.5 },
+        { "monster": "GROUP_COWS", "x": [ 1, 15 ], "y": [ 1, 12 ], "density": 0.5 }
```

Comment:
0.5 density only hurts the cows more.

I dont personally know the exact effect density has on mons, but setting it to 1 or 5 definitely highlights that it does have some effect.  Leave density out alltogether and/or specify it in the terrain.json entry?